File: src/main/java/com/clearcont/clearcontapp/service/EmpresaGroupService.java
package com.clearcont.clearcontapp.service;

import com.clearcont.clearcontapp.model.EmpresaGroup;
import com.clearcont.clearcontapp.repository.EmpresaGroupRepository;
import org.springframework.stereotype.Service;

@Service
public class EmpresaGroupService {
    private final EmpresaGroupRepository repository;
    
    public EmpresaGroupService(EmpresaGroupRepository repository) {
        this.repository = repository;
    }
    
    public EmpresaGroup getByID(Long id) {
        return repository.findById(id).orElseThrow();
    }
    
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/service/EmpresaGroupService.kt
package br.com.clearcont.clearcontwebapp.service

import br.com.clearcont.clearcontwebapp.models.EmpresaGroup
import br.com.clearcont.clearcontwebapp.repository.EmpresaGroupRepository
import org.springframework.stereotype.Service


@Service
class EmpresaGroupService(private val repository: EmpresaGroupRepository) {
    fun getByID(id: Long): EmpresaGroup {
        return repository.findById(id).orElseThrow()
    }

    fun update(empresaGroup: EmpresaGroup): EmpresaGroup {
        return repository.saveAndFlush(empresaGroup)
    }
}
